Major updates for v3 (#4)
DocOps/liquidoc-cmf: README.adoc

. *Run Bundler* to install dependencies.
+
[source,shell]
----
bundle
----
+
If an error indicates Bundler is not installed, run `gem install bundler`, then repeat this command.

. *Rename this README* so the upcoming _init_ operation does not overwrite it.
+
.Example
[source,shell]
----
mv README.adoc README.meta.adoc
----
+
This file may confuse future users; we advise removing it before sharing your LDCMF project.

. *Edit the metafile* for your new LDCMF project.
+
In your favorite text/code editor, open `data/meta.yml` and fill out the required info, then *save the file*.

. *Initialize your custom environment.*
[lowerroman]
.. Generate an _initialization config_.
+
[source,shell]
----
bundle exec liquidoc --template _init/templates/init.yaml --data data/meta.yml --output _init/init.yml
----
+
This command invokes barest LiquiDoc, which performs a one-off template parsing.

.. Generate _infrastructure files_ based on your custom preferences.
+
[source,shell]
----
bundle exec liquidoc -c _init/init.yml
----

=== Build the Docs

. You can now *review the generated `README.adoc`*. +

> The new README file is oriented for _your LDCMF application's_ future users (and you!).

Reviewer: I would also suggest we add a comment at the top of the generated README that explains what `docs_build_command_basic` is doing there. As a new user, that would confuse me.

. *Perform the first build of your seedling docset*, as instructed in the new README. +

> 

Reviewer: I'm a bit confused here, and I'm not sure if it's just because I'm testing.

This what I specified in `meta.yml`....which of course created the corresponding instructions in the new README. 
![image](https://user-images.githubusercontent.com/30160806/47524058-6d717180-d84e-11e8-947b-45e253e61275.png)

But it didn't actually create that new repo. I'm still in my bootstrap-test repo on my local machine. So following the README instructions isn't going to take me anywhere. What am I missing?

Reviewer: Also, as a user I am wondering how I will regenerate these docs for my documentation project as time goes on.

Author: That's it. You're IN your new documentation project. This part is hard to explain but it's easier than simple to do -- it's already done. The whole objective is to convert the cloned repo into your new documentation project. You don't need to be "taken anywhere" :-)

=== Wrapup

> . When you're certain everything looks good, *delete the `_init/` directory.

Reviewer: - As a new user, how specifically can I feel comfortable that everything looks good?
- And how do I go check it out?
- Also, do we want to include a build and serve command instead, and then instruct users to check that address?
- Let's also specifically call out the fact that it automatically generates a PDF and an HTML version of the docs for this doc project.
- Missing end `*`

Author: There is no serve yet. The Jekyll part is rudimentary, because there are just too many ways that could be implemented and I haven't generalized it yet. It preps a lot of the Jekyll but it doesn't have render actions configured, so there's nothing to serve. That will probably come in v4.

The rest is all great points. I'll make it more explicit.

= LiquiDoc CMF
// This AsciiDoc file must be rendered to be properly viewed.
// The easiest way to view it rendered is on GitHub at
// https://github.com/DocOps/liquidoc-cmf
// OR copy and paste these contents into
// https://asciidoclive.com

> // BELOW is all AsciiDoc formatting:

Reviewer: `Everything below these comments is in Asciidoc syntax.`

This file usually contains just simple data: a big (or small) column of basic key-value pairs to create simple variables.

data/products.yrml:::
For subdivided information about your products in distinct blocks.
Each block can be called for selective ingest during build routines using the colon signifier, such as by calling `data/products.yml:product-1`, where `product-1:` is a top-level block in the `products.yml` file.

data/guides.yml:::
This block is for content-oriented data that is distinct between the different portals or guides you're producing.
This is often redundant to your `products.yml` file, if product editions themselves are the major point of divergence in your docs, and it is formatted the same way.
For _this project_ (LDCMF Guides), the _guides_ are oriented toward _audiences_ (documentarians, admins, and developers), but the products (LiquiDoc and LDCMF) are distinct from this and actually documented/instructed _together_ in each guide.
+
[TIP]
Favoring the filename `products.yml` is conventional when products and guides (portals) have a 1:1 relationship and `guides.yml` file is superfluous.

data/schema.yml:::

> Can also be `data/manifest.yml`, this crucial file provides, central manifest of all page-forming content items (pages, topics) and how they are organized (metadata such as categories into which content items fall).

Reviewer: ```suggestion
This crucial file provides a central manifest of all page-forming content items (pages, topics). It dictates how they are organized in the site navigation (using metadata such as categories into which content items fall). Can also be `data/manifest.yml`.
```

> The schema file carries essential build info that lets us see relationships between topics and build content-exclusive portals from otherwise fairly dumb, decontextualized repositories.

Reviewer: ```suggestion
The schema file carries essential build info that lets us see relationships between topics and build content-exclusive portals from otherwise-decontextualized repositories.
```

data/terms.yml:::
By no means a required file, `terms.yml` is a great example of a file that is really just for content.

> You can have as many of these key-value files, serving whatever purposes you may wish.

Reviewer: ```suggestion
You can have as many of these key-value files, serving whatever purposes you wish.
```

products/::
This is an optional path for LDCMF projects.

> If you plan to embed your product repos as submodules, the `products/` directory is the base path to stick them in.

Reviewer: ```suggestion
If you plan to embed your product repos as submodules, put them in the `products/` directory.
```

> For LDCMF Guides, this path effectively leads to symlinks for the LiquiDoc and LDCMF repos, so any files within those repos are accessible to be drawn into our docs.

Reviewer: Is the symlink thing also true, in theory, for other projects that use submodules? If so, let's generalize that a bit more.

This mainly includes Jekyll templates (`themes/<theme-name>/_includes/` and `themes/<theme-name>/_layouts/`) and asset files such as stylesheets, front-end javascripts, and of course theme-related images.
This would also be the home of PDF and slideshow output theme configurations, as applicable.

theme/pdf-theme.yml::
A very basic PDF theming file based on Asciidoctor's link:https://github.com/asciidoctor/asciidoctor-pdf/blob/master/data/themes/default-theme.yml[*default-theme.yml*], just to get you started.
For more about PDF theming, see the link:https://github.com/asciidoctor/asciidoctor-pdf/blob/master/docs/theming-guide.adoc[Asciidoctor PDF Theming Guide].
// end::architecture-descriptions[]

== Build Config

The `_configs/build-docs.yml` file is the brains of any LDCMF application.
It defines the sequential compilation routine and ensures all assets are in place for the final artifact rendering operations.

=== LDCMF Build Strategies
// tag::build-strategies[]
LiquiDoc CMF isn't much without its ability to maintain strictly “DRY” single sourcing while still producing diverse output.

Store small data in flat files.::

> Most of the data about products is not best stored in relational databases -- it is not used live, schemas mostly get in the way, and version control is essential.

Reviewer: This sentence confuses me...

If you want to keep your product info in Git, use appropriate flat-file formats for various representations.
The human-friendliest formats are probably YAML and CSV.

> YAML can be edited in any decent code editor, and CSV can be edited in a plethora of spreadsheet applications.

Reviewer: ```suggestion
YAML can be edited in any decent code editor, and a CSV file can be edited in any spreadsheet application.
```

Share product data with the product.::

> Ideally, that small data we just agreed to store as flat files should be kept in one canonical place, from which it is read by the docs and the product build routines, ensuring docs references reflect the current truth about the product.

Reviewer: ```suggestion
Ideally, that small data stored in flat files should be kept in one canonical place from which it is read by the docs and the product build routines, ensuring docs references reflect a single current source of truth about the product.
```

Prebuild and include complex reference content.::
All that small data needs to make it into your docs in a more human-readable format.
This is where prebuilding reference content to AsciiDoc source comes in.
Use Liquid templates to generate includable AsciiDoc files into the `_build/snippets/` directory.
Then include them into your static AsciiDoc files.

Handle major divergence by configuring output into “guides”.::

> Major points of divergence in output needs -- such as significantly different “editions” of the same product or highly variant audiences, like basic vs advanced users or consumers vs developers -- each necessitates its own _guide_.

Reviewer: ```suggestion
When there are major points of divergence in output -- such as significantly different “editions” of the same product or highly variant audiences, like basic vs advanced users or consumers vs developers -- each necessitates its own _guide_.
```

Guides are built sequentially, each drawing configuration settings and content designated for it, along with content and data common to other guides.
This process generates parallel guides, including Web portals that are built side by side and served as components of one site.

Handle output-format diversity with “portals” and “manuals”.::
A lot of the conflict in documentation output stems from the manuals vs portals debate.
Modern websites tend to work best by presenting content in semi-serialized or unserialized article format, more like Wikipedia than a book.
Meanwhile, technical documentation is often still intended to be consumed more like a book or a traditional manual.

> LDCMF tries to balance both without requiring either, but each final rendering operation is technically building a book-style _manual_ or a help-site/wiki-style _portal_, even if you are eventually deploying both.

Reviewer: Is this saying that we have to choose when we build?

bundle exec liquidoc --template _init/templates/init.yaml --data data/meta.yml --output _init/init.yml
----
+
This command performs a one-off template parsing to establish the temporary _init_ config file, responding with a message that your `_init/init.yml` was built.

.. Generate _infrastructure files_ based on your custom preferences.
+
[source,shell]
----
bundle exec liquidoc -c _init/init.yml
----
+
This procedure builds starter files (and inflates directories), including a new `README.adoc` and `content/index.adoc`.

> If your initialization steps proceeded without errors, you should have a successful rudimentary LDCMF codebase.

Reviewer: Maybe `you will have a basic LDCMF codebase.`

bundle exec liquidoc -c _init/init.yml
----
+
This procedure builds starter files (and inflates directories).

> If your initialization steps proceeded without errors, you now have a successful rudimentary LDCMF codebase, starting with a `_configs/` directory containing LiquiDoc, Asciidoctor, and Jekyll config files, as well as new `README.adoc` and `content/index.adoc` files.

Reviewer: Oh this is good, except I'd avoid using the term 'rudimentary'. I just put this in the other review, but what about:

If your initialization steps proceeded without error, they generate a basic codebase that serves as the framework for a LDCMF documentation project, including a `_configs/` directory containing LiquiDoc, Asciidoctor, and Jekyll config files, as well as new `README.adoc` and `content/index.adoc` files.

It defines the sequential compilation routine and ensures all assets are in place for the final artifact rendering operations.

=== LDCMF Build Strategies
// tag::build-strategies[]

> LiquiDoc CMF isn't much without its ability to maintain strictly “DRY” single sourcing while still producing diverse output.

Reviewer: `The strength of  LiquiDoc CMF is its ability to maintain strictly “DRY” single sourcing while still producing diverse output.`

Reviewer: @briandominick did you see this one?
